streams/src/test/java/org/apache/kafka/streams/processor/internals/DefaultStateUpdaterTest.java

```diff
@@ -125,6 +125,80 @@ public void shouldShutdownStateUpdaterAndRestart() {
         verify(changelogReader, times(2)).clear();
     }
 
+    @Test
+    public void shouldRemoveTasksFromAndClearInputQueueOnShutdown() throws Exception {
+        stateUpdater.shutdown(Duration.ofMinutes(1));
```

Comment:
Why we call `shutdown` first? We should have shutdown upon tearing down each test.

Ditto in `shouldRemoveUpdatingTasksOnShutdown`.

Reply:
That is basically just a safe guard in case somebody modifies the test setup in the future. For example, if somebody starts the state updater in a `setup()` method. This shutdown ensures that we have a clean and not-started state updater for this test. 

In `shouldRemoveUpdatingTasksOnShutdown()` it ensures that we shutdown the state updater created on line 89 before we re-create it with a very high commit interval so that we can be sure that the checkpoint that is created was created by the clean-up code and not during normal state updating.

Reply:
> That is basically just a safe guard in case somebody modifies the test setup in the future.

I see. In that case why do we not do the same in other test cases?